{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "orthologs = pd.read_csv('/cndd/Public_Datasets/single_cell_methylome/summary_files/orthologs.txt',\n",
    "                        sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "evals = pd.read_csv('/cndd2/ethan/projects/enh_gene_linkage/intergenic_corr/data/evals_w_fdr_100kb_12k_genes.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>gene</th>\n",
       "      <th>ens</th>\n",
       "      <th>in_gene</th>\n",
       "      <th>val</th>\n",
       "      <th>val2</th>\n",
       "      <th>val_clustershuffled</th>\n",
       "      <th>val2_clustershuffled</th>\n",
       "      <th>dist</th>\n",
       "      <th>pvalues</th>\n",
       "      <th>...</th>\n",
       "      <th>fdr&lt;100kb shuffled (cluster label)</th>\n",
       "      <th>pvalues2</th>\n",
       "      <th>fdr2&lt;100kb</th>\n",
       "      <th>fdr2&lt;200kb</th>\n",
       "      <th>fdr2&lt;500kb</th>\n",
       "      <th>fdr2&lt;100kb shuffled (cluster label)</th>\n",
       "      <th>fdr_mcg</th>\n",
       "      <th>fdr_mcg_corr</th>\n",
       "      <th>fdr_atac</th>\n",
       "      <th>fdr_atac_corr</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>94</td>\n",
       "      <td>ENSMUSG00000000001</td>\n",
       "      <td>147662</td>\n",
       "      <td>False</td>\n",
       "      <td>-0.039720</td>\n",
       "      <td>0.044097</td>\n",
       "      <td>-0.265784</td>\n",
       "      <td>-0.136448</td>\n",
       "      <td>92177</td>\n",
       "      <td>0.424733</td>\n",
       "      <td>...</td>\n",
       "      <td>1.095838</td>\n",
       "      <td>0.450781</td>\n",
       "      <td>0.744850</td>\n",
       "      <td>0.797258</td>\n",
       "      <td>0.876520</td>\n",
       "      <td>1.032819</td>\n",
       "      <td>0.726151</td>\n",
       "      <td>0.662643</td>\n",
       "      <td>0.744850</td>\n",
       "      <td>0.721180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>95</td>\n",
       "      <td>ENSMUSG00000000001</td>\n",
       "      <td>147663</td>\n",
       "      <td>False</td>\n",
       "      <td>0.024620</td>\n",
       "      <td>0.051756</td>\n",
       "      <td>-0.380239</td>\n",
       "      <td>-0.101215</td>\n",
       "      <td>91955</td>\n",
       "      <td>0.499490</td>\n",
       "      <td>...</td>\n",
       "      <td>0.922071</td>\n",
       "      <td>0.439489</td>\n",
       "      <td>0.738874</td>\n",
       "      <td>0.792315</td>\n",
       "      <td>0.873352</td>\n",
       "      <td>1.051270</td>\n",
       "      <td>0.770029</td>\n",
       "      <td>0.835107</td>\n",
       "      <td>0.738874</td>\n",
       "      <td>0.702838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>96</td>\n",
       "      <td>ENSMUSG00000000001</td>\n",
       "      <td>147664</td>\n",
       "      <td>False</td>\n",
       "      <td>-0.022212</td>\n",
       "      <td>0.074516</td>\n",
       "      <td>-0.081081</td>\n",
       "      <td>-0.112376</td>\n",
       "      <td>84155</td>\n",
       "      <td>0.445065</td>\n",
       "      <td>...</td>\n",
       "      <td>1.037449</td>\n",
       "      <td>0.406090</td>\n",
       "      <td>0.720486</td>\n",
       "      <td>0.776735</td>\n",
       "      <td>0.862787</td>\n",
       "      <td>1.112977</td>\n",
       "      <td>0.738378</td>\n",
       "      <td>0.711723</td>\n",
       "      <td>0.720486</td>\n",
       "      <td>0.647348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>97</td>\n",
       "      <td>ENSMUSG00000000001</td>\n",
       "      <td>147665</td>\n",
       "      <td>False</td>\n",
       "      <td>-0.147171</td>\n",
       "      <td>0.017398</td>\n",
       "      <td>0.194660</td>\n",
       "      <td>-0.279352</td>\n",
       "      <td>76112</td>\n",
       "      <td>0.306660</td>\n",
       "      <td>...</td>\n",
       "      <td>1.762511</td>\n",
       "      <td>0.490344</td>\n",
       "      <td>0.765055</td>\n",
       "      <td>0.813756</td>\n",
       "      <td>0.886827</td>\n",
       "      <td>0.979848</td>\n",
       "      <td>0.648328</td>\n",
       "      <td>0.367841</td>\n",
       "      <td>0.765055</td>\n",
       "      <td>0.780787</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>98</td>\n",
       "      <td>ENSMUSG00000000001</td>\n",
       "      <td>147666</td>\n",
       "      <td>False</td>\n",
       "      <td>0.122880</td>\n",
       "      <td>0.122005</td>\n",
       "      <td>0.050224</td>\n",
       "      <td>0.120910</td>\n",
       "      <td>63221</td>\n",
       "      <td>0.611919</td>\n",
       "      <td>...</td>\n",
       "      <td>0.806321</td>\n",
       "      <td>0.338683</td>\n",
       "      <td>0.679892</td>\n",
       "      <td>0.742359</td>\n",
       "      <td>0.840268</td>\n",
       "      <td>1.285390</td>\n",
       "      <td>0.829016</td>\n",
       "      <td>1.028145</td>\n",
       "      <td>0.679892</td>\n",
       "      <td>0.528936</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0                gene     ens  in_gene       val      val2  \\\n",
       "0          94  ENSMUSG00000000001  147662    False -0.039720  0.044097   \n",
       "1          95  ENSMUSG00000000001  147663    False  0.024620  0.051756   \n",
       "2          96  ENSMUSG00000000001  147664    False -0.022212  0.074516   \n",
       "3          97  ENSMUSG00000000001  147665    False -0.147171  0.017398   \n",
       "4          98  ENSMUSG00000000001  147666    False  0.122880  0.122005   \n",
       "\n",
       "   val_clustershuffled  val2_clustershuffled   dist   pvalues  ...  \\\n",
       "0            -0.265784             -0.136448  92177  0.424733  ...   \n",
       "1            -0.380239             -0.101215  91955  0.499490  ...   \n",
       "2            -0.081081             -0.112376  84155  0.445065  ...   \n",
       "3             0.194660             -0.279352  76112  0.306660  ...   \n",
       "4             0.050224              0.120910  63221  0.611919  ...   \n",
       "\n",
       "   fdr<100kb shuffled (cluster label)  pvalues2  fdr2<100kb  fdr2<200kb  \\\n",
       "0                            1.095838  0.450781    0.744850    0.797258   \n",
       "1                            0.922071  0.439489    0.738874    0.792315   \n",
       "2                            1.037449  0.406090    0.720486    0.776735   \n",
       "3                            1.762511  0.490344    0.765055    0.813756   \n",
       "4                            0.806321  0.338683    0.679892    0.742359   \n",
       "\n",
       "   fdr2<500kb  fdr2<100kb shuffled (cluster label)   fdr_mcg  fdr_mcg_corr  \\\n",
       "0    0.876520                             1.032819  0.726151      0.662643   \n",
       "1    0.873352                             1.051270  0.770029      0.835107   \n",
       "2    0.862787                             1.112977  0.738378      0.711723   \n",
       "3    0.886827                             0.979848  0.648328      0.367841   \n",
       "4    0.840268                             1.285390  0.829016      1.028145   \n",
       "\n",
       "   fdr_atac  fdr_atac_corr  \n",
       "0  0.744850       0.721180  \n",
       "1  0.738874       0.702838  \n",
       "2  0.720486       0.647348  \n",
       "3  0.765055       0.780787  \n",
       "4  0.679892       0.528936  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "evals.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "atac_genes = evals.loc[evals.fdr_atac_corr < .05].gene.unique()\n",
    "mcg_genes = evals.loc[evals.fdr_mcg_corr < .05].gene.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8071"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(atac_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10732"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(mcg_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "possible_genes = np.union1d(atac_genes, mcg_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11014"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(possible_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneType</th>\n",
       "      <th>hsa_gID</th>\n",
       "      <th>hsa_tID</th>\n",
       "      <th>hsa_pID</th>\n",
       "      <th>hsa_gName</th>\n",
       "      <th>hsa_tName</th>\n",
       "      <th>mmu_gID</th>\n",
       "      <th>mmu_tID</th>\n",
       "      <th>mmu_pID</th>\n",
       "      <th>mmu_gName</th>\n",
       "      <th>mmu_tName</th>\n",
       "      <th>HumanPercId</th>\n",
       "      <th>MousePercId</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENSG00000136928</td>\n",
       "      <td>ENST00000259455</td>\n",
       "      <td>ENSP00000259455</td>\n",
       "      <td>GABBR2</td>\n",
       "      <td>GABBR2-001</td>\n",
       "      <td>ENSMUSG00000039809</td>\n",
       "      <td>ENSMUST00000107749</td>\n",
       "      <td>ENSMUSP00000103378</td>\n",
       "      <td>Gabbr2</td>\n",
       "      <td>Gabbr2-001</td>\n",
       "      <td>97.0</td>\n",
       "      <td>97.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENSG00000140931</td>\n",
       "      <td>ENST00000424011</td>\n",
       "      <td>ENSP00000400482</td>\n",
       "      <td>CMTM3</td>\n",
       "      <td>CMTM3-001</td>\n",
       "      <td>ENSMUSG00000031875</td>\n",
       "      <td>ENSMUST00000034343</td>\n",
       "      <td>ENSMUSP00000034343</td>\n",
       "      <td>Cmtm3</td>\n",
       "      <td>Cmtm3-201</td>\n",
       "      <td>88.0</td>\n",
       "      <td>88.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENSG00000123472</td>\n",
       "      <td>ENST00000371937</td>\n",
       "      <td>ENSP00000361005</td>\n",
       "      <td>ATPAF1</td>\n",
       "      <td>ATPAF1-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSMUSP00000030478</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Atpaf1-001</td>\n",
       "      <td>89.0</td>\n",
       "      <td>89.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENSG00000125780</td>\n",
       "      <td>ENST00000381458</td>\n",
       "      <td>ENSP00000370867</td>\n",
       "      <td>TGM3</td>\n",
       "      <td>TGM3-001</td>\n",
       "      <td>ENSMUSG00000027401</td>\n",
       "      <td>ENSMUST00000110299</td>\n",
       "      <td>ENSMUSP00000105928</td>\n",
       "      <td>Tgm3</td>\n",
       "      <td>Tgm3-001</td>\n",
       "      <td>76.0</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENSG00000224531</td>\n",
       "      <td>ENST00000416247</td>\n",
       "      <td>ENSP00000451866</td>\n",
       "      <td>SMIM13</td>\n",
       "      <td>C6orf228-002</td>\n",
       "      <td>ENSMUSG00000091264</td>\n",
       "      <td>ENSMUST00000165561</td>\n",
       "      <td>ENSMUSP00000130956</td>\n",
       "      <td>Smim13</td>\n",
       "      <td>BC024659-001</td>\n",
       "      <td>88.0</td>\n",
       "      <td>88.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  GeneType          hsa_gID          hsa_tID          hsa_pID hsa_gName  \\\n",
       "0  protein  ENSG00000136928  ENST00000259455  ENSP00000259455    GABBR2   \n",
       "1  protein  ENSG00000140931  ENST00000424011  ENSP00000400482     CMTM3   \n",
       "2  protein  ENSG00000123472  ENST00000371937  ENSP00000361005    ATPAF1   \n",
       "3  protein  ENSG00000125780  ENST00000381458  ENSP00000370867      TGM3   \n",
       "4  protein  ENSG00000224531  ENST00000416247  ENSP00000451866    SMIM13   \n",
       "\n",
       "      hsa_tName             mmu_gID             mmu_tID             mmu_pID  \\\n",
       "0    GABBR2-001  ENSMUSG00000039809  ENSMUST00000107749  ENSMUSP00000103378   \n",
       "1     CMTM3-001  ENSMUSG00000031875  ENSMUST00000034343  ENSMUSP00000034343   \n",
       "2    ATPAF1-001                 NaN                 NaN  ENSMUSP00000030478   \n",
       "3      TGM3-001  ENSMUSG00000027401  ENSMUST00000110299  ENSMUSP00000105928   \n",
       "4  C6orf228-002  ENSMUSG00000091264  ENSMUST00000165561  ENSMUSP00000130956   \n",
       "\n",
       "  mmu_gName     mmu_tName  HumanPercId  MousePercId  \n",
       "0    Gabbr2    Gabbr2-001         97.0         97.0  \n",
       "1     Cmtm3     Cmtm3-201         88.0         88.0  \n",
       "2       NaN    Atpaf1-001         89.0         89.0  \n",
       "3      Tgm3      Tgm3-001         76.0         76.0  \n",
       "4    Smim13  BC024659-001         88.0         88.0  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orthologs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "orthologs = orthologs.loc[orthologs.mmu_gID.isin(possible_genes)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(9054, 13)"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orthologs.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneType</th>\n",
       "      <th>hsa_gID</th>\n",
       "      <th>hsa_tID</th>\n",
       "      <th>hsa_pID</th>\n",
       "      <th>hsa_gName</th>\n",
       "      <th>hsa_tName</th>\n",
       "      <th>mmu_gID</th>\n",
       "      <th>mmu_tID</th>\n",
       "      <th>mmu_pID</th>\n",
       "      <th>mmu_gName</th>\n",
       "      <th>mmu_tName</th>\n",
       "      <th>HumanPercId</th>\n",
       "      <th>MousePercId</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENSG00000136928</td>\n",
       "      <td>ENST00000259455</td>\n",
       "      <td>ENSP00000259455</td>\n",
       "      <td>GABBR2</td>\n",
       "      <td>GABBR2-001</td>\n",
       "      <td>ENSMUSG00000039809</td>\n",
       "      <td>ENSMUST00000107749</td>\n",
       "      <td>ENSMUSP00000103378</td>\n",
       "      <td>Gabbr2</td>\n",
       "      <td>Gabbr2-001</td>\n",
       "      <td>97.0</td>\n",
       "      <td>97.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENSG00000224531</td>\n",
       "      <td>ENST00000416247</td>\n",
       "      <td>ENSP00000451866</td>\n",
       "      <td>SMIM13</td>\n",
       "      <td>C6orf228-002</td>\n",
       "      <td>ENSMUSG00000091264</td>\n",
       "      <td>ENSMUST00000165561</td>\n",
       "      <td>ENSMUSP00000130956</td>\n",
       "      <td>Smim13</td>\n",
       "      <td>BC024659-001</td>\n",
       "      <td>88.0</td>\n",
       "      <td>88.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENSG00000161981</td>\n",
       "      <td>ENST00000383018</td>\n",
       "      <td>ENSP00000372482</td>\n",
       "      <td>SNRNP25</td>\n",
       "      <td>SNRNP25-201</td>\n",
       "      <td>ENSMUSG00000040767</td>\n",
       "      <td>ENSMUST00000039601</td>\n",
       "      <td>ENSMUSP00000046654</td>\n",
       "      <td>Snrnp25</td>\n",
       "      <td>Snrnp25-001</td>\n",
       "      <td>98.0</td>\n",
       "      <td>98.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENSG00000125741</td>\n",
       "      <td>ENST00000323060</td>\n",
       "      <td>ENSP00000319817</td>\n",
       "      <td>OPA3</td>\n",
       "      <td>OPA3-202</td>\n",
       "      <td>ENSMUSG00000052214</td>\n",
       "      <td>ENSMUST00000063976</td>\n",
       "      <td>ENSMUSP00000069965</td>\n",
       "      <td>Opa3</td>\n",
       "      <td>Opa3-001</td>\n",
       "      <td>76.0</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENSG00000204104</td>\n",
       "      <td>ENST00000373327</td>\n",
       "      <td>ENSP00000362424</td>\n",
       "      <td>TRAF3IP1</td>\n",
       "      <td>TRAF3IP1-001</td>\n",
       "      <td>ENSMUSG00000034292</td>\n",
       "      <td>ENSMUST00000047242</td>\n",
       "      <td>ENSMUSP00000042391</td>\n",
       "      <td>Traf3ip1</td>\n",
       "      <td>Traf3ip1-201</td>\n",
       "      <td>70.0</td>\n",
       "      <td>70.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   GeneType          hsa_gID          hsa_tID          hsa_pID hsa_gName  \\\n",
       "0   protein  ENSG00000136928  ENST00000259455  ENSP00000259455    GABBR2   \n",
       "4   protein  ENSG00000224531  ENST00000416247  ENSP00000451866    SMIM13   \n",
       "8   protein  ENSG00000161981  ENST00000383018  ENSP00000372482   SNRNP25   \n",
       "11  protein  ENSG00000125741  ENST00000323060  ENSP00000319817      OPA3   \n",
       "12  protein  ENSG00000204104  ENST00000373327  ENSP00000362424  TRAF3IP1   \n",
       "\n",
       "       hsa_tName             mmu_gID             mmu_tID             mmu_pID  \\\n",
       "0     GABBR2-001  ENSMUSG00000039809  ENSMUST00000107749  ENSMUSP00000103378   \n",
       "4   C6orf228-002  ENSMUSG00000091264  ENSMUST00000165561  ENSMUSP00000130956   \n",
       "8    SNRNP25-201  ENSMUSG00000040767  ENSMUST00000039601  ENSMUSP00000046654   \n",
       "11      OPA3-202  ENSMUSG00000052214  ENSMUST00000063976  ENSMUSP00000069965   \n",
       "12  TRAF3IP1-001  ENSMUSG00000034292  ENSMUST00000047242  ENSMUSP00000042391   \n",
       "\n",
       "   mmu_gName     mmu_tName  HumanPercId  MousePercId  \n",
       "0     Gabbr2    Gabbr2-001         97.0         97.0  \n",
       "4     Smim13  BC024659-001         88.0         88.0  \n",
       "8    Snrnp25   Snrnp25-001         98.0         98.0  \n",
       "11      Opa3      Opa3-001         76.0         76.0  \n",
       "12  Traf3ip1  Traf3ip1-201         70.0         70.0  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orthologs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8040"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.sum(orthologs.MousePercId > 75)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "orthologs = orthologs.loc[orthologs.MousePercId > 75]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# get human genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "hg38 = pd.read_csv('/cndd/junhao/genomes/hg38/gencode.v25.annotation.gtf',\n",
    "                   skiprows = 5, sep='\\t', header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>11869</td>\n",
       "      <td>14409</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000223972.5\"; gene_type \"transc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>11869</td>\n",
       "      <td>14409</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000223972.5\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>11869</td>\n",
       "      <td>12227</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000223972.5\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>12613</td>\n",
       "      <td>12721</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000223972.5\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>13221</td>\n",
       "      <td>14409</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000223972.5\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      0       1           2      3      4  5  6  7  \\\n",
       "0  chr1  HAVANA        gene  11869  14409  .  +  .   \n",
       "1  chr1  HAVANA  transcript  11869  14409  .  +  .   \n",
       "2  chr1  HAVANA        exon  11869  12227  .  +  .   \n",
       "3  chr1  HAVANA        exon  12613  12721  .  +  .   \n",
       "4  chr1  HAVANA        exon  13221  14409  .  +  .   \n",
       "\n",
       "                                                   8  \n",
       "0  gene_id \"ENSG00000223972.5\"; gene_type \"transc...  \n",
       "1  gene_id \"ENSG00000223972.5\"; transcript_id \"EN...  \n",
       "2  gene_id \"ENSG00000223972.5\"; transcript_id \"EN...  \n",
       "3  gene_id \"ENSG00000223972.5\"; transcript_id \"EN...  \n",
       "4  gene_id \"ENSG00000223972.5\"; transcript_id \"EN...  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hg38.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "hg38 = hg38.loc[hg38[2] == 'gene']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>11869</td>\n",
       "      <td>14409</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000223972.5\"; gene_type \"transc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>14404</td>\n",
       "      <td>29570</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000227232.5\"; gene_type \"unproc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>gene</td>\n",
       "      <td>17369</td>\n",
       "      <td>17436</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000278267.1\"; gene_type \"miRNA\"...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>29554</td>\n",
       "      <td>31109</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000243485.4\"; gene_type \"lincRN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>34554</td>\n",
       "      <td>36081</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000237613.2\"; gene_type \"lincRN...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       0        1     2      3      4  5  6  7  \\\n",
       "0   chr1   HAVANA  gene  11869  14409  .  +  .   \n",
       "12  chr1   HAVANA  gene  14404  29570  .  -  .   \n",
       "25  chr1  ENSEMBL  gene  17369  17436  .  -  .   \n",
       "28  chr1   HAVANA  gene  29554  31109  .  +  .   \n",
       "38  chr1   HAVANA  gene  34554  36081  .  -  .   \n",
       "\n",
       "                                                    8  \n",
       "0   gene_id \"ENSG00000223972.5\"; gene_type \"transc...  \n",
       "12  gene_id \"ENSG00000227232.5\"; gene_type \"unproc...  \n",
       "25  gene_id \"ENSG00000278267.1\"; gene_type \"miRNA\"...  \n",
       "28  gene_id \"ENSG00000243485.4\"; gene_type \"lincRN...  \n",
       "38  gene_id \"ENSG00000237613.2\"; gene_type \"lincRN...  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hg38.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "hg38 = hg38.drop(8, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "hg38.to_csv('hg38_gtf_cleaned', index=None, columns=None, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>gene_status</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>11869</td>\n",
       "      <td>14409</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>\"transcribed_unprocessed_pseudogene\"</td>\n",
       "      <td>\"KNOWN\"</td>\n",
       "      <td>\"DDX11L1\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>14404</td>\n",
       "      <td>29570</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>\"unprocessed_pseudogene\"</td>\n",
       "      <td>\"KNOWN\"</td>\n",
       "      <td>\"WASH7P\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>gene</td>\n",
       "      <td>17369</td>\n",
       "      <td>17436</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000278267</td>\n",
       "      <td>\"miRNA\"</td>\n",
       "      <td>\"KNOWN\"</td>\n",
       "      <td>\"MIR6859-1\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>29554</td>\n",
       "      <td>31109</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000243485</td>\n",
       "      <td>\"lincRNA\"</td>\n",
       "      <td>\"KNOWN\"</td>\n",
       "      <td>\"MIR1302-2\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>34554</td>\n",
       "      <td>36081</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000237613</td>\n",
       "      <td>\"lincRNA\"</td>\n",
       "      <td>\"KNOWN\"</td>\n",
       "      <td>\"FAM138A\"</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       0        1     2      3      4  5  6  7          gene_id  \\\n",
       "0   chr1   HAVANA  gene  11869  14409  .  +  .  ENSG00000223972   \n",
       "12  chr1   HAVANA  gene  14404  29570  .  -  .  ENSG00000227232   \n",
       "25  chr1  ENSEMBL  gene  17369  17436  .  -  .  ENSG00000278267   \n",
       "28  chr1   HAVANA  gene  29554  31109  .  +  .  ENSG00000243485   \n",
       "38  chr1   HAVANA  gene  34554  36081  .  -  .  ENSG00000237613   \n",
       "\n",
       "                               gene_type gene_status    gene_name  \n",
       "0   \"transcribed_unprocessed_pseudogene\"     \"KNOWN\"    \"DDX11L1\"  \n",
       "12              \"unprocessed_pseudogene\"     \"KNOWN\"     \"WASH7P\"  \n",
       "25                               \"miRNA\"     \"KNOWN\"  \"MIR6859-1\"  \n",
       "28                             \"lincRNA\"     \"KNOWN\"  \"MIR1302-2\"  \n",
       "38                             \"lincRNA\"     \"KNOWN\"    \"FAM138A\"  "
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hg38.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "hg38['gene_id'] = [i.strip('\"') for i in hg38['gene_id']]\n",
    "hg38['gene_id'] = [i.split('.')[0] for i in hg38['gene_id']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneType</th>\n",
       "      <th>hsa_gID</th>\n",
       "      <th>hsa_tID</th>\n",
       "      <th>hsa_pID</th>\n",
       "      <th>hsa_gName</th>\n",
       "      <th>hsa_tName</th>\n",
       "      <th>mmu_gID</th>\n",
       "      <th>mmu_tID</th>\n",
       "      <th>mmu_pID</th>\n",
       "      <th>mmu_gName</th>\n",
       "      <th>mmu_tName</th>\n",
       "      <th>HumanPercId</th>\n",
       "      <th>MousePercId</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENSG00000136928</td>\n",
       "      <td>ENST00000259455</td>\n",
       "      <td>ENSP00000259455</td>\n",
       "      <td>GABBR2</td>\n",
       "      <td>GABBR2-001</td>\n",
       "      <td>ENSMUSG00000039809</td>\n",
       "      <td>ENSMUST00000107749</td>\n",
       "      <td>ENSMUSP00000103378</td>\n",
       "      <td>Gabbr2</td>\n",
       "      <td>Gabbr2-001</td>\n",
       "      <td>97.0</td>\n",
       "      <td>97.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENSG00000224531</td>\n",
       "      <td>ENST00000416247</td>\n",
       "      <td>ENSP00000451866</td>\n",
       "      <td>SMIM13</td>\n",
       "      <td>C6orf228-002</td>\n",
       "      <td>ENSMUSG00000091264</td>\n",
       "      <td>ENSMUST00000165561</td>\n",
       "      <td>ENSMUSP00000130956</td>\n",
       "      <td>Smim13</td>\n",
       "      <td>BC024659-001</td>\n",
       "      <td>88.0</td>\n",
       "      <td>88.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENSG00000161981</td>\n",
       "      <td>ENST00000383018</td>\n",
       "      <td>ENSP00000372482</td>\n",
       "      <td>SNRNP25</td>\n",
       "      <td>SNRNP25-201</td>\n",
       "      <td>ENSMUSG00000040767</td>\n",
       "      <td>ENSMUST00000039601</td>\n",
       "      <td>ENSMUSP00000046654</td>\n",
       "      <td>Snrnp25</td>\n",
       "      <td>Snrnp25-001</td>\n",
       "      <td>98.0</td>\n",
       "      <td>98.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENSG00000125741</td>\n",
       "      <td>ENST00000323060</td>\n",
       "      <td>ENSP00000319817</td>\n",
       "      <td>OPA3</td>\n",
       "      <td>OPA3-202</td>\n",
       "      <td>ENSMUSG00000052214</td>\n",
       "      <td>ENSMUST00000063976</td>\n",
       "      <td>ENSMUSP00000069965</td>\n",
       "      <td>Opa3</td>\n",
       "      <td>Opa3-001</td>\n",
       "      <td>76.0</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENSG00000015592</td>\n",
       "      <td>ENST00000350889</td>\n",
       "      <td>ENSP00000342538</td>\n",
       "      <td>STMN4</td>\n",
       "      <td>STMN4-001</td>\n",
       "      <td>ENSMUSG00000022044</td>\n",
       "      <td>ENSMUST00000074523</td>\n",
       "      <td>ENSMUSP00000074113</td>\n",
       "      <td>Stmn4</td>\n",
       "      <td>Stmn4-001</td>\n",
       "      <td>87.0</td>\n",
       "      <td>87.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   GeneType          hsa_gID          hsa_tID          hsa_pID hsa_gName  \\\n",
       "0   protein  ENSG00000136928  ENST00000259455  ENSP00000259455    GABBR2   \n",
       "4   protein  ENSG00000224531  ENST00000416247  ENSP00000451866    SMIM13   \n",
       "8   protein  ENSG00000161981  ENST00000383018  ENSP00000372482   SNRNP25   \n",
       "11  protein  ENSG00000125741  ENST00000323060  ENSP00000319817      OPA3   \n",
       "14  protein  ENSG00000015592  ENST00000350889  ENSP00000342538     STMN4   \n",
       "\n",
       "       hsa_tName             mmu_gID             mmu_tID             mmu_pID  \\\n",
       "0     GABBR2-001  ENSMUSG00000039809  ENSMUST00000107749  ENSMUSP00000103378   \n",
       "4   C6orf228-002  ENSMUSG00000091264  ENSMUST00000165561  ENSMUSP00000130956   \n",
       "8    SNRNP25-201  ENSMUSG00000040767  ENSMUST00000039601  ENSMUSP00000046654   \n",
       "11      OPA3-202  ENSMUSG00000052214  ENSMUST00000063976  ENSMUSP00000069965   \n",
       "14     STMN4-001  ENSMUSG00000022044  ENSMUST00000074523  ENSMUSP00000074113   \n",
       "\n",
       "   mmu_gName     mmu_tName  HumanPercId  MousePercId  \n",
       "0     Gabbr2    Gabbr2-001         97.0         97.0  \n",
       "4     Smim13  BC024659-001         88.0         88.0  \n",
       "8    Snrnp25   Snrnp25-001         98.0         98.0  \n",
       "11      Opa3      Opa3-001         76.0         76.0  \n",
       "14     Stmn4     Stmn4-001         87.0         87.0  "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orthologs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneType</th>\n",
       "      <th>hsa_gID</th>\n",
       "      <th>hsa_tID</th>\n",
       "      <th>hsa_pID</th>\n",
       "      <th>hsa_gName</th>\n",
       "      <th>hsa_tName</th>\n",
       "      <th>mmu_gID</th>\n",
       "      <th>mmu_tID</th>\n",
       "      <th>mmu_pID</th>\n",
       "      <th>mmu_gName</th>\n",
       "      <th>mmu_tName</th>\n",
       "      <th>HumanPercId</th>\n",
       "      <th>MousePercId</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [GeneType, hsa_gID, hsa_tID, hsa_pID, hsa_gName, hsa_tName, mmu_gID, mmu_tID, mmu_pID, mmu_gName, mmu_tName, HumanPercId, MousePercId]\n",
       "Index: []"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orthologs.loc[orthologs.mmu_gName == \"Stim2\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "use_genes = hg38.loc[hg38.gene_id.isin(orthologs['hsa_gID'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(7763, 12)"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "use_genes.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(8040, 13)"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orthologs.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>gene_status</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1465</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>1020123</td>\n",
       "      <td>1056118</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000188157</td>\n",
       "      <td>\"protein_coding\"</td>\n",
       "      <td>\"KNOWN\"</td>\n",
       "      <td>\"AGRN\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2396</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>1253909</td>\n",
       "      <td>1273885</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000160087</td>\n",
       "      <td>\"protein_coding\"</td>\n",
       "      <td>\"KNOWN\"</td>\n",
       "      <td>\"UBE2J2\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2936</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>1292376</td>\n",
       "      <td>1309609</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000131584</td>\n",
       "      <td>\"protein_coding\"</td>\n",
       "      <td>\"KNOWN\"</td>\n",
       "      <td>\"ACAP3\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3193</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>1311585</td>\n",
       "      <td>1324691</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000127054</td>\n",
       "      <td>\"protein_coding\"</td>\n",
       "      <td>\"KNOWN\"</td>\n",
       "      <td>\"CPSF3L\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4045</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>1324756</td>\n",
       "      <td>1328897</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000224051</td>\n",
       "      <td>\"protein_coding\"</td>\n",
       "      <td>\"KNOWN\"</td>\n",
       "      <td>\"CPTP\"</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         0       1     2        3        4  5  6  7          gene_id  \\\n",
       "1465  chr1  HAVANA  gene  1020123  1056118  .  +  .  ENSG00000188157   \n",
       "2396  chr1  HAVANA  gene  1253909  1273885  .  -  .  ENSG00000160087   \n",
       "2936  chr1  HAVANA  gene  1292376  1309609  .  -  .  ENSG00000131584   \n",
       "3193  chr1  HAVANA  gene  1311585  1324691  .  -  .  ENSG00000127054   \n",
       "4045  chr1  HAVANA  gene  1324756  1328897  .  +  .  ENSG00000224051   \n",
       "\n",
       "             gene_type gene_status gene_name  \n",
       "1465  \"protein_coding\"     \"KNOWN\"    \"AGRN\"  \n",
       "2396  \"protein_coding\"     \"KNOWN\"  \"UBE2J2\"  \n",
       "2936  \"protein_coding\"     \"KNOWN\"   \"ACAP3\"  \n",
       "3193  \"protein_coding\"     \"KNOWN\"  \"CPSF3L\"  \n",
       "4045  \"protein_coding\"     \"KNOWN\"    \"CPTP\"  "
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "use_genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_tss(gtf_df):\n",
    "    tss = [] \n",
    "    for row in gtf_df.itertuples():\n",
    "        \n",
    "        if row[7] == '-':\n",
    "            tss.append(row[5])\n",
    "        else:\n",
    "            tss.append(row[4])\n",
    "    print(row)\n",
    "    return tss"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pandas(Index=2572696, _1='chrY', _2='HAVANA', _3='gene', _4=6910686, _5=7091683, _6='.', _7='+', _8='.', gene_id='ENSG00000092377', gene_type='\"protein_coding\"', gene_status='\"KNOWN\"', gene_name='\"TBL1Y\"')\n"
     ]
    }
   ],
   "source": [
    "tss_genes = get_tss(use_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneType</th>\n",
       "      <th>hsa_gID</th>\n",
       "      <th>hsa_tID</th>\n",
       "      <th>hsa_pID</th>\n",
       "      <th>hsa_gName</th>\n",
       "      <th>hsa_tName</th>\n",
       "      <th>mmu_gID</th>\n",
       "      <th>mmu_tID</th>\n",
       "      <th>mmu_pID</th>\n",
       "      <th>mmu_gName</th>\n",
       "      <th>mmu_tName</th>\n",
       "      <th>HumanPercId</th>\n",
       "      <th>MousePercId</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENSG00000136928</td>\n",
       "      <td>ENST00000259455</td>\n",
       "      <td>ENSP00000259455</td>\n",
       "      <td>GABBR2</td>\n",
       "      <td>GABBR2-001</td>\n",
       "      <td>ENSMUSG00000039809</td>\n",
       "      <td>ENSMUST00000107749</td>\n",
       "      <td>ENSMUSP00000103378</td>\n",
       "      <td>Gabbr2</td>\n",
       "      <td>Gabbr2-001</td>\n",
       "      <td>97.0</td>\n",
       "      <td>97.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENSG00000224531</td>\n",
       "      <td>ENST00000416247</td>\n",
       "      <td>ENSP00000451866</td>\n",
       "      <td>SMIM13</td>\n",
       "      <td>C6orf228-002</td>\n",
       "      <td>ENSMUSG00000091264</td>\n",
       "      <td>ENSMUST00000165561</td>\n",
       "      <td>ENSMUSP00000130956</td>\n",
       "      <td>Smim13</td>\n",
       "      <td>BC024659-001</td>\n",
       "      <td>88.0</td>\n",
       "      <td>88.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENSG00000161981</td>\n",
       "      <td>ENST00000383018</td>\n",
       "      <td>ENSP00000372482</td>\n",
       "      <td>SNRNP25</td>\n",
       "      <td>SNRNP25-201</td>\n",
       "      <td>ENSMUSG00000040767</td>\n",
       "      <td>ENSMUST00000039601</td>\n",
       "      <td>ENSMUSP00000046654</td>\n",
       "      <td>Snrnp25</td>\n",
       "      <td>Snrnp25-001</td>\n",
       "      <td>98.0</td>\n",
       "      <td>98.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENSG00000125741</td>\n",
       "      <td>ENST00000323060</td>\n",
       "      <td>ENSP00000319817</td>\n",
       "      <td>OPA3</td>\n",
       "      <td>OPA3-202</td>\n",
       "      <td>ENSMUSG00000052214</td>\n",
       "      <td>ENSMUST00000063976</td>\n",
       "      <td>ENSMUSP00000069965</td>\n",
       "      <td>Opa3</td>\n",
       "      <td>Opa3-001</td>\n",
       "      <td>76.0</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENSG00000015592</td>\n",
       "      <td>ENST00000350889</td>\n",
       "      <td>ENSP00000342538</td>\n",
       "      <td>STMN4</td>\n",
       "      <td>STMN4-001</td>\n",
       "      <td>ENSMUSG00000022044</td>\n",
       "      <td>ENSMUST00000074523</td>\n",
       "      <td>ENSMUSP00000074113</td>\n",
       "      <td>Stmn4</td>\n",
       "      <td>Stmn4-001</td>\n",
       "      <td>87.0</td>\n",
       "      <td>87.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   GeneType          hsa_gID          hsa_tID          hsa_pID hsa_gName  \\\n",
       "0   protein  ENSG00000136928  ENST00000259455  ENSP00000259455    GABBR2   \n",
       "4   protein  ENSG00000224531  ENST00000416247  ENSP00000451866    SMIM13   \n",
       "8   protein  ENSG00000161981  ENST00000383018  ENSP00000372482   SNRNP25   \n",
       "11  protein  ENSG00000125741  ENST00000323060  ENSP00000319817      OPA3   \n",
       "14  protein  ENSG00000015592  ENST00000350889  ENSP00000342538     STMN4   \n",
       "\n",
       "       hsa_tName             mmu_gID             mmu_tID             mmu_pID  \\\n",
       "0     GABBR2-001  ENSMUSG00000039809  ENSMUST00000107749  ENSMUSP00000103378   \n",
       "4   C6orf228-002  ENSMUSG00000091264  ENSMUST00000165561  ENSMUSP00000130956   \n",
       "8    SNRNP25-201  ENSMUSG00000040767  ENSMUST00000039601  ENSMUSP00000046654   \n",
       "11      OPA3-202  ENSMUSG00000052214  ENSMUST00000063976  ENSMUSP00000069965   \n",
       "14     STMN4-001  ENSMUSG00000022044  ENSMUST00000074523  ENSMUSP00000074113   \n",
       "\n",
       "   mmu_gName     mmu_tName  HumanPercId  MousePercId  \n",
       "0     Gabbr2    Gabbr2-001         97.0         97.0  \n",
       "4     Smim13  BC024659-001         88.0         88.0  \n",
       "8    Snrnp25   Snrnp25-001         98.0         98.0  \n",
       "11      Opa3      Opa3-001         76.0         76.0  \n",
       "14     Stmn4     Stmn4-001         87.0         87.0  "
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orthologs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "orthologs = orthologs.set_index(\"hsa_gID\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneType</th>\n",
       "      <th>hsa_tID</th>\n",
       "      <th>hsa_pID</th>\n",
       "      <th>hsa_gName</th>\n",
       "      <th>hsa_tName</th>\n",
       "      <th>mmu_gID</th>\n",
       "      <th>mmu_tID</th>\n",
       "      <th>mmu_pID</th>\n",
       "      <th>mmu_gName</th>\n",
       "      <th>mmu_tName</th>\n",
       "      <th>HumanPercId</th>\n",
       "      <th>MousePercId</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hsa_gID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000136928</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENST00000259455</td>\n",
       "      <td>ENSP00000259455</td>\n",
       "      <td>GABBR2</td>\n",
       "      <td>GABBR2-001</td>\n",
       "      <td>ENSMUSG00000039809</td>\n",
       "      <td>ENSMUST00000107749</td>\n",
       "      <td>ENSMUSP00000103378</td>\n",
       "      <td>Gabbr2</td>\n",
       "      <td>Gabbr2-001</td>\n",
       "      <td>97.0</td>\n",
       "      <td>97.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000224531</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENST00000416247</td>\n",
       "      <td>ENSP00000451866</td>\n",
       "      <td>SMIM13</td>\n",
       "      <td>C6orf228-002</td>\n",
       "      <td>ENSMUSG00000091264</td>\n",
       "      <td>ENSMUST00000165561</td>\n",
       "      <td>ENSMUSP00000130956</td>\n",
       "      <td>Smim13</td>\n",
       "      <td>BC024659-001</td>\n",
       "      <td>88.0</td>\n",
       "      <td>88.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000161981</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENST00000383018</td>\n",
       "      <td>ENSP00000372482</td>\n",
       "      <td>SNRNP25</td>\n",
       "      <td>SNRNP25-201</td>\n",
       "      <td>ENSMUSG00000040767</td>\n",
       "      <td>ENSMUST00000039601</td>\n",
       "      <td>ENSMUSP00000046654</td>\n",
       "      <td>Snrnp25</td>\n",
       "      <td>Snrnp25-001</td>\n",
       "      <td>98.0</td>\n",
       "      <td>98.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000125741</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENST00000323060</td>\n",
       "      <td>ENSP00000319817</td>\n",
       "      <td>OPA3</td>\n",
       "      <td>OPA3-202</td>\n",
       "      <td>ENSMUSG00000052214</td>\n",
       "      <td>ENSMUST00000063976</td>\n",
       "      <td>ENSMUSP00000069965</td>\n",
       "      <td>Opa3</td>\n",
       "      <td>Opa3-001</td>\n",
       "      <td>76.0</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000015592</th>\n",
       "      <td>protein</td>\n",
       "      <td>ENST00000350889</td>\n",
       "      <td>ENSP00000342538</td>\n",
       "      <td>STMN4</td>\n",
       "      <td>STMN4-001</td>\n",
       "      <td>ENSMUSG00000022044</td>\n",
       "      <td>ENSMUST00000074523</td>\n",
       "      <td>ENSMUSP00000074113</td>\n",
       "      <td>Stmn4</td>\n",
       "      <td>Stmn4-001</td>\n",
       "      <td>87.0</td>\n",
       "      <td>87.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                GeneType          hsa_tID          hsa_pID hsa_gName  \\\n",
       "hsa_gID                                                                \n",
       "ENSG00000136928  protein  ENST00000259455  ENSP00000259455    GABBR2   \n",
       "ENSG00000224531  protein  ENST00000416247  ENSP00000451866    SMIM13   \n",
       "ENSG00000161981  protein  ENST00000383018  ENSP00000372482   SNRNP25   \n",
       "ENSG00000125741  protein  ENST00000323060  ENSP00000319817      OPA3   \n",
       "ENSG00000015592  protein  ENST00000350889  ENSP00000342538     STMN4   \n",
       "\n",
       "                    hsa_tName             mmu_gID             mmu_tID  \\\n",
       "hsa_gID                                                                 \n",
       "ENSG00000136928    GABBR2-001  ENSMUSG00000039809  ENSMUST00000107749   \n",
       "ENSG00000224531  C6orf228-002  ENSMUSG00000091264  ENSMUST00000165561   \n",
       "ENSG00000161981   SNRNP25-201  ENSMUSG00000040767  ENSMUST00000039601   \n",
       "ENSG00000125741      OPA3-202  ENSMUSG00000052214  ENSMUST00000063976   \n",
       "ENSG00000015592     STMN4-001  ENSMUSG00000022044  ENSMUST00000074523   \n",
       "\n",
       "                            mmu_pID mmu_gName     mmu_tName  HumanPercId  \\\n",
       "hsa_gID                                                                    \n",
       "ENSG00000136928  ENSMUSP00000103378    Gabbr2    Gabbr2-001         97.0   \n",
       "ENSG00000224531  ENSMUSP00000130956    Smim13  BC024659-001         88.0   \n",
       "ENSG00000161981  ENSMUSP00000046654   Snrnp25   Snrnp25-001         98.0   \n",
       "ENSG00000125741  ENSMUSP00000069965      Opa3      Opa3-001         76.0   \n",
       "ENSG00000015592  ENSMUSP00000074113     Stmn4     Stmn4-001         87.0   \n",
       "\n",
       "                 MousePercId  \n",
       "hsa_gID                       \n",
       "ENSG00000136928         97.0  \n",
       "ENSG00000224531         88.0  \n",
       "ENSG00000161981         98.0  \n",
       "ENSG00000125741         76.0  \n",
       "ENSG00000015592         87.0  "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orthologs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "use_gene_bed = pd.DataFrame(columns = ['chr', 'start_tss', 'end_tss', 'human_gid', 'mouse_gid'],\n",
    "                        data=np.array([use_genes[0], np.array(tss_genes) -2000,\n",
    "                              np.array(tss_genes)+2000, use_genes['gene_id'],\n",
    "                              orthologs.loc[use_genes['gene_id'], \"mmu_gID\"]\n",
    "                             ]).T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start_tss</th>\n",
       "      <th>end_tss</th>\n",
       "      <th>human_gid</th>\n",
       "      <th>mouse_gid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1018123</td>\n",
       "      <td>1022123</td>\n",
       "      <td>ENSG00000188157</td>\n",
       "      <td>ENSMUSG00000041936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1271885</td>\n",
       "      <td>1275885</td>\n",
       "      <td>ENSG00000160087</td>\n",
       "      <td>ENSMUSG00000023286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1307609</td>\n",
       "      <td>1311609</td>\n",
       "      <td>ENSG00000131584</td>\n",
       "      <td>ENSMUSG00000029033</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1322691</td>\n",
       "      <td>1326691</td>\n",
       "      <td>ENSG00000127054</td>\n",
       "      <td>ENSMUSG00000029034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1322756</td>\n",
       "      <td>1326756</td>\n",
       "      <td>ENSG00000224051</td>\n",
       "      <td>ENSMUSG00000029073</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr start_tss  end_tss        human_gid           mouse_gid\n",
       "0  chr1   1018123  1022123  ENSG00000188157  ENSMUSG00000041936\n",
       "1  chr1   1271885  1275885  ENSG00000160087  ENSMUSG00000023286\n",
       "2  chr1   1307609  1311609  ENSG00000131584  ENSMUSG00000029033\n",
       "3  chr1   1322691  1326691  ENSG00000127054  ENSMUSG00000029034\n",
       "4  chr1   1322756  1326756  ENSG00000224051  ENSMUSG00000029073"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "use_gene_bed.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\u001b[0;31mSignature:\u001b[0m\n",
       "\u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mto_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\u001b[0m\n",
       "\u001b[0;34m\u001b[0m    \u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
       "\u001b[0;34m\u001b[0m    \u001b[0mpath_or_buf\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
       "\u001b[0;34m\u001b[0m    \u001b[0msep\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m','\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
       "\u001b[0;34m\u001b[0m    \u001b[0mna_rep\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m''\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
       "\u001b[0;34m\u001b[0m    \u001b[0mfloat_format\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
       "\u001b[0;34m\u001b[0m    \u001b[0mcolumns\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
       "\u001b[0;34m\u001b[0m    \u001b[0mheader\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
       "\u001b[0;34m\u001b[0m    \u001b[0mindex\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
       "\u001b[0;34m\u001b[0m    \u001b[0mindex_label\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
       "\u001b[0;34m\u001b[0m    \u001b[0mmode\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'w'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
       "\u001b[0;34m\u001b[0m    \u001b[0mencoding\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
       "\u001b[0;34m\u001b[0m    \u001b[0mcompression\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'infer'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
       "\u001b[0;34m\u001b[0m    \u001b[0mquoting\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
       "\u001b[0;34m\u001b[0m    \u001b[0mquotechar\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'\"'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
       "\u001b[0;34m\u001b[0m    \u001b[0mline_terminator\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
       "\u001b[0;34m\u001b[0m    \u001b[0mchunksize\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
       "\u001b[0;34m\u001b[0m    \u001b[0mtupleize_cols\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
       "\u001b[0;34m\u001b[0m    \u001b[0mdate_format\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
       "\u001b[0;34m\u001b[0m    \u001b[0mdoublequote\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
       "\u001b[0;34m\u001b[0m    \u001b[0mescapechar\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
       "\u001b[0;34m\u001b[0m    \u001b[0mdecimal\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'.'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
       "\u001b[0;34m\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
       "\u001b[0;31mDocstring:\u001b[0m\n",
       "Write object to a comma-separated values (csv) file.\n",
       "\n",
       ".. versionchanged:: 0.24.0\n",
       "    The order of arguments for Series was changed.\n",
       "\n",
       "Parameters\n",
       "----------\n",
       "path_or_buf : str or file handle, default None\n",
       "    File path or object, if None is provided the result is returned as\n",
       "    a string.  If a file object is passed it should be opened with\n",
       "    `newline=''`, disabling universal newlines.\n",
       "\n",
       "    .. versionchanged:: 0.24.0\n",
       "\n",
       "       Was previously named \"path\" for Series.\n",
       "\n",
       "sep : str, default ','\n",
       "    String of length 1. Field delimiter for the output file.\n",
       "na_rep : str, default ''\n",
       "    Missing data representation.\n",
       "float_format : str, default None\n",
       "    Format string for floating point numbers.\n",
       "columns : sequence, optional\n",
       "    Columns to write.\n",
       "header : bool or list of str, default True\n",
       "    Write out the column names. If a list of strings is given it is\n",
       "    assumed to be aliases for the column names.\n",
       "\n",
       "    .. versionchanged:: 0.24.0\n",
       "\n",
       "       Previously defaulted to False for Series.\n",
       "\n",
       "index : bool, default True\n",
       "    Write row names (index).\n",
       "index_label : str or sequence, or False, default None\n",
       "    Column label for index column(s) if desired. If None is given, and\n",
       "    `header` and `index` are True, then the index names are used. A\n",
       "    sequence should be given if the object uses MultiIndex. If\n",
       "    False do not print fields for index names. Use index_label=False\n",
       "    for easier importing in R.\n",
       "mode : str\n",
       "    Python write mode, default 'w'.\n",
       "encoding : str, optional\n",
       "    A string representing the encoding to use in the output file,\n",
       "    defaults to 'ascii' on Python 2 and 'utf-8' on Python 3.\n",
       "compression : str, default 'infer'\n",
       "    Compression mode among the following possible values: {'infer',\n",
       "    'gzip', 'bz2', 'zip', 'xz', None}. If 'infer' and `path_or_buf`\n",
       "    is path-like, then detect compression from the following\n",
       "    extensions: '.gz', '.bz2', '.zip' or '.xz'. (otherwise no\n",
       "    compression).\n",
       "\n",
       "    .. versionchanged:: 0.24.0\n",
       "\n",
       "       'infer' option added and set to default.\n",
       "\n",
       "quoting : optional constant from csv module\n",
       "    Defaults to csv.QUOTE_MINIMAL. If you have set a `float_format`\n",
       "    then floats are converted to strings and thus csv.QUOTE_NONNUMERIC\n",
       "    will treat them as non-numeric.\n",
       "quotechar : str, default '\\\"'\n",
       "    String of length 1. Character used to quote fields.\n",
       "line_terminator : string, optional\n",
       "    The newline character or character sequence to use in the output\n",
       "    file. Defaults to `os.linesep`, which depends on the OS in which\n",
       "    this method is called ('\\n' for linux, '\\r\\n' for Windows, i.e.).\n",
       "\n",
       "    .. versionchanged:: 0.24.0\n",
       "chunksize : int or None\n",
       "    Rows to write at a time.\n",
       "tupleize_cols : bool, default False\n",
       "    Write MultiIndex columns as a list of tuples (if True) or in\n",
       "    the new, expanded format, where each MultiIndex column is a row\n",
       "    in the CSV (if False).\n",
       "\n",
       "    .. deprecated:: 0.21.0\n",
       "       This argument will be removed and will always write each row\n",
       "       of the multi-index as a separate row in the CSV file.\n",
       "date_format : str, default None\n",
       "    Format string for datetime objects.\n",
       "doublequote : bool, default True\n",
       "    Control quoting of `quotechar` inside a field.\n",
       "escapechar : str, default None\n",
       "    String of length 1. Character used to escape `sep` and `quotechar`\n",
       "    when appropriate.\n",
       "decimal : str, default '.'\n",
       "    Character recognized as decimal separator. E.g. use ',' for\n",
       "    European data.\n",
       "\n",
       "Returns\n",
       "-------\n",
       "None or str\n",
       "    If path_or_buf is None, returns the resulting csv format as a\n",
       "    string. Otherwise returns None.\n",
       "\n",
       "See Also\n",
       "--------\n",
       "read_csv : Load a CSV file into a DataFrame.\n",
       "to_excel : Load an Excel file into a DataFrame.\n",
       "\n",
       "Examples\n",
       "--------\n",
       ">>> df = pd.DataFrame({'name': ['Raphael', 'Donatello'],\n",
       "...                    'mask': ['red', 'purple'],\n",
       "...                    'weapon': ['sai', 'bo staff']})\n",
       ">>> df.to_csv(index=False)\n",
       "'name,mask,weapon\\nRaphael,red,sai\\nDonatello,purple,bo staff\\n'\n",
       "\u001b[0;31mFile:\u001b[0m      /cndd/earmand/conda_envs/multiplex/lib/python3.7/site-packages/pandas/core/generic.py\n",
       "\u001b[0;31mType:\u001b[0m      function\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "?pd.DataFrame.to_csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "use_gene_bed.to_csv('use_gene_bed', sep='\\t', header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "test = pd.read_csv('use_gene_bed', sep='\\t', header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1018123</td>\n",
       "      <td>1022123</td>\n",
       "      <td>ENSG00000188157</td>\n",
       "      <td>ENSMUSG00000041936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1271885</td>\n",
       "      <td>1275885</td>\n",
       "      <td>ENSG00000160087</td>\n",
       "      <td>ENSMUSG00000023286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1307609</td>\n",
       "      <td>1311609</td>\n",
       "      <td>ENSG00000131584</td>\n",
       "      <td>ENSMUSG00000029033</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1322691</td>\n",
       "      <td>1326691</td>\n",
       "      <td>ENSG00000127054</td>\n",
       "      <td>ENSMUSG00000029034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1322756</td>\n",
       "      <td>1326756</td>\n",
       "      <td>ENSG00000224051</td>\n",
       "      <td>ENSMUSG00000029073</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      0        1        2                3                   4\n",
       "0  chr1  1018123  1022123  ENSG00000188157  ENSMUSG00000041936\n",
       "1  chr1  1271885  1275885  ENSG00000160087  ENSMUSG00000023286\n",
       "2  chr1  1307609  1311609  ENSG00000131584  ENSMUSG00000029033\n",
       "3  chr1  1322691  1326691  ENSG00000127054  ENSMUSG00000029034\n",
       "4  chr1  1322756  1326756  ENSG00000224051  ENSMUSG00000029073"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "use_gene_bed['chr'] = use_genes[0]\n",
    "use_gene_bed['start_tss'] = tss_genes - 2000\n",
    "use_gene_bed['end_tss'] = tss_genes + 2000\n",
    "use_gene_bed['human_gid'] = use_genes['gene_id']\n",
    "use_gene_bed['mouse_gid'] = use_genes['mmu_gID']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "?str.strip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "to_split = hg38[8]\n",
    "splitted = [i.split(';') for i in to_split]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['gene_id \"ENSG00000223972.5\"',\n",
       " ' gene_type \"transcribed_unprocessed_pseudogene\"',\n",
       " ' gene_status \"KNOWN\"',\n",
       " ' gene_name \"DDX11L1\"',\n",
       " ' level 2',\n",
       " ' havana_gene \"OTTHUMG00000000961.2\"',\n",
       " '']"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "splitted[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_id = []\n",
    "gene_type = []\n",
    "gene_status = []\n",
    "gene_name = []\n",
    "gene_level = []\n",
    "for strings in splitted:\n",
    "    subsplit = [i.split(' ') for i in strings]\n",
    "    gene_id.append(subsplit[0][1])\n",
    "    gene_type.append(subsplit[1][2])\n",
    "    gene_status.append(subsplit[2][2])\n",
    "    gene_name.append(subsplit[3][2])\n",
    "    gene_level.append(subsplit[4][2])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'splitted_2' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-24-c305ed33ffd6>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0msplitted_2\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'splitted_2' is not defined"
     ]
    }
   ],
   "source": [
    "splitted_2[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "hg38['gene_id'] = gene_id\n",
    "hg38['gene_type'] = gene_type\n",
    "hg38['gene_status'] = gene_status\n",
    "hg38['gene_name'] = gene_name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>gene_status</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>11869</td>\n",
       "      <td>14409</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000223972.5\"; gene_type \"transc...</td>\n",
       "      <td>\"ENSG00000223972.5\"</td>\n",
       "      <td>\"transcribed_unprocessed_pseudogene\"</td>\n",
       "      <td>\"KNOWN\"</td>\n",
       "      <td>\"DDX11L1\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>14404</td>\n",
       "      <td>29570</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000227232.5\"; gene_type \"unproc...</td>\n",
       "      <td>\"ENSG00000227232.5\"</td>\n",
       "      <td>\"unprocessed_pseudogene\"</td>\n",
       "      <td>\"KNOWN\"</td>\n",
       "      <td>\"WASH7P\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>gene</td>\n",
       "      <td>17369</td>\n",
       "      <td>17436</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000278267.1\"; gene_type \"miRNA\"...</td>\n",
       "      <td>\"ENSG00000278267.1\"</td>\n",
       "      <td>\"miRNA\"</td>\n",
       "      <td>\"KNOWN\"</td>\n",
       "      <td>\"MIR6859-1\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>29554</td>\n",
       "      <td>31109</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000243485.4\"; gene_type \"lincRN...</td>\n",
       "      <td>\"ENSG00000243485.4\"</td>\n",
       "      <td>\"lincRNA\"</td>\n",
       "      <td>\"KNOWN\"</td>\n",
       "      <td>\"MIR1302-2\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>34554</td>\n",
       "      <td>36081</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000237613.2\"; gene_type \"lincRN...</td>\n",
       "      <td>\"ENSG00000237613.2\"</td>\n",
       "      <td>\"lincRNA\"</td>\n",
       "      <td>\"KNOWN\"</td>\n",
       "      <td>\"FAM138A\"</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       0        1     2      3      4  5  6  7  \\\n",
       "0   chr1   HAVANA  gene  11869  14409  .  +  .   \n",
       "12  chr1   HAVANA  gene  14404  29570  .  -  .   \n",
       "25  chr1  ENSEMBL  gene  17369  17436  .  -  .   \n",
       "28  chr1   HAVANA  gene  29554  31109  .  +  .   \n",
       "38  chr1   HAVANA  gene  34554  36081  .  -  .   \n",
       "\n",
       "                                                    8              gene_id  \\\n",
       "0   gene_id \"ENSG00000223972.5\"; gene_type \"transc...  \"ENSG00000223972.5\"   \n",
       "12  gene_id \"ENSG00000227232.5\"; gene_type \"unproc...  \"ENSG00000227232.5\"   \n",
       "25  gene_id \"ENSG00000278267.1\"; gene_type \"miRNA\"...  \"ENSG00000278267.1\"   \n",
       "28  gene_id \"ENSG00000243485.4\"; gene_type \"lincRN...  \"ENSG00000243485.4\"   \n",
       "38  gene_id \"ENSG00000237613.2\"; gene_type \"lincRN...  \"ENSG00000237613.2\"   \n",
       "\n",
       "                               gene_type gene_status    gene_name  \n",
       "0   \"transcribed_unprocessed_pseudogene\"     \"KNOWN\"    \"DDX11L1\"  \n",
       "12              \"unprocessed_pseudogene\"     \"KNOWN\"     \"WASH7P\"  \n",
       "25                               \"miRNA\"     \"KNOWN\"  \"MIR6859-1\"  \n",
       "28                             \"lincRNA\"     \"KNOWN\"  \"MIR1302-2\"  \n",
       "38                             \"lincRNA\"     \"KNOWN\"    \"FAM138A\"  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hg38.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "splitted_2 = [i.split(' ') for i in ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "splitted_2 = []\n",
    "for i in splitted:\n",
    "    splitted_2.append([j.split(' ') for j in i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['gene_id', '\"ENSG00000223972.5\"']"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "splitted_2[0][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "f = open('/cndd2/Public_Datasets/references/hg19/transcriptome/gencode.v19.annotation.gtf', 'r')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "check = f.readline()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'##description: evidence-based annotation of the human genome (GRCh37), version 19 (Ensembl 74)\\n'"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "hg19.head()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
