{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "sys.path.insert(0, '/cndd/fangming/CEMBA/snmcseq_dev')\n",
    "sys.path.insert(0, '/cndd/fangming/CEMBA/scripts/ClusterCrossValidation')\n",
    "\n",
    "from __init__ import *\n",
    "from __init__jupyterlab import *\n",
    "import seaborn as sns\n",
    "set_matplotlib_formats('retina')\n",
    "import importlib\n",
    "import snmcseq_utils\n",
    "importlib.reload(snmcseq_utils)\n",
    "from matplotlib import collections as mc\n",
    "\n",
    "\n",
    "import pickle\n",
    "import xarray\n",
    "import fbpca\n",
    "from sklearn.model_selection import KFold\n",
    "from sklearn.metrics import adjusted_rand_score\n",
    "from sklearn.metrics import r2_score\n",
    "import itertools\n",
    "from scipy import sparse\n",
    "from scipy.stats import spearmanr\n",
    "from scipy import optimize\n",
    "from scipy.optimize import root_scalar\n",
    "from scipy import stats\n",
    "from statsmodels.stats.multitest import multipletests\n",
    "\n",
    "import CEMBA_run_tsne\n",
    "import CEMBA_clst_utils\n",
    "import cluster_cv_utils\n",
    "importlib.reload(cluster_cv_utils)\n",
    "from cluster_cv_utils import *\n",
    "\n",
    "import gseapy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1771112, 9)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>3073253</td>\n",
       "      <td>3074322</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000102693.1\"; gene_type \"TEC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>3073253</td>\n",
       "      <td>3074322</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000102693.1\"; transcript_id ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>3073253</td>\n",
       "      <td>3074322</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000102693.1\"; transcript_id ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>gene</td>\n",
       "      <td>3102016</td>\n",
       "      <td>3102125</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000064842.1\"; gene_type \"snR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>transcript</td>\n",
       "      <td>3102016</td>\n",
       "      <td>3102125</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000064842.1\"; transcript_id ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      0        1           2        3        4  5  6  7  \\\n",
       "0  chr1   HAVANA        gene  3073253  3074322  .  +  .   \n",
       "1  chr1   HAVANA  transcript  3073253  3074322  .  +  .   \n",
       "2  chr1   HAVANA        exon  3073253  3074322  .  +  .   \n",
       "3  chr1  ENSEMBL        gene  3102016  3102125  .  +  .   \n",
       "4  chr1  ENSEMBL  transcript  3102016  3102125  .  +  .   \n",
       "\n",
       "                                                   8  \n",
       "0  gene_id \"ENSMUSG00000102693.1\"; gene_type \"TEC...  \n",
       "1  gene_id \"ENSMUSG00000102693.1\"; transcript_id ...  \n",
       "2  gene_id \"ENSMUSG00000102693.1\"; transcript_id ...  \n",
       "3  gene_id \"ENSMUSG00000064842.1\"; gene_type \"snR...  \n",
       "4  gene_id \"ENSMUSG00000064842.1\"; transcript_id ...  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f = '/cndd2/fangming/references/gencode.vM16.annotation.gtf.gz'\n",
    "df = pd.read_csv(f, sep='\\t', skiprows=5, header=None)\n",
    "print(df.shape)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['gene', 'transcript', 'exon', 'CDS', 'start_codon', 'stop_codon',\n",
       "       'UTR', 'Selenocysteine'], dtype=object)"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[2].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(133849, 9)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>3073253</td>\n",
       "      <td>3074322</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000102693.1\"; transcript_id ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>transcript</td>\n",
       "      <td>3102016</td>\n",
       "      <td>3102125</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000064842.1\"; transcript_id ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>3205901</td>\n",
       "      <td>3216344</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000051951.5\"; transcript_id ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>3206523</td>\n",
       "      <td>3215632</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000051951.5\"; transcript_id ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>3214482</td>\n",
       "      <td>3671498</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000051951.5\"; transcript_id ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       0        1           2        3        4  5  6  7  \\\n",
       "1   chr1   HAVANA  transcript  3073253  3074322  .  +  .   \n",
       "4   chr1  ENSEMBL  transcript  3102016  3102125  .  +  .   \n",
       "7   chr1   HAVANA  transcript  3205901  3216344  .  -  .   \n",
       "10  chr1   HAVANA  transcript  3206523  3215632  .  -  .   \n",
       "13  chr1   HAVANA  transcript  3214482  3671498  .  -  .   \n",
       "\n",
       "                                                    8  \n",
       "1   gene_id \"ENSMUSG00000102693.1\"; transcript_id ...  \n",
       "4   gene_id \"ENSMUSG00000064842.1\"; transcript_id ...  \n",
       "7   gene_id \"ENSMUSG00000051951.5\"; transcript_id ...  \n",
       "10  gene_id \"ENSMUSG00000051951.5\"; transcript_id ...  \n",
       "13  gene_id \"ENSMUSG00000051951.5\"; transcript_id ...  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df[df[2]=='transcript']\n",
    "print(df.shape)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'gene_id \"ENSMUSG00000102693.1\"; transcript_id \"ENSMUST00000193812.1\"; gene_type \"TEC\"; gene_name \"4933401J01Rik\"; transcript_type \"TEC\"; transcript_name \"4933401J01Rik-201\"; level 2; transcript_support_level \"NA\"; tag \"basic\"; havana_gene \"OTTMUSG00000049935.1\"; havana_transcript \"OTTMUST00000127109.1\";'"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.iloc[0, 8]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "df['gene_id'] = df[8].apply(lambda x: x.split(';')[0].split('\"')[1])\n",
    "df['transcript_id'] = df[8].apply(lambda x: x.split(';')[1].split('\"')[1])\n",
    "df['gene_name'] = df[8].apply(lambda x: x.split(';')[3].split('\"')[1])\n",
    "df['transcript_name'] = df[8].apply(lambda x: x.split(';')[5].split('\"')[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>3073253</td>\n",
       "      <td>3074322</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000102693.1\"; transcript_id ...</td>\n",
       "      <td>ENSMUSG00000102693.1</td>\n",
       "      <td>ENSMUST00000193812.1</td>\n",
       "      <td>4933401J01Rik</td>\n",
       "      <td>4933401J01Rik-201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>transcript</td>\n",
       "      <td>3102016</td>\n",
       "      <td>3102125</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000064842.1\"; transcript_id ...</td>\n",
       "      <td>ENSMUSG00000064842.1</td>\n",
       "      <td>ENSMUST00000082908.1</td>\n",
       "      <td>Gm26206</td>\n",
       "      <td>Gm26206-201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>3205901</td>\n",
       "      <td>3216344</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000051951.5\"; transcript_id ...</td>\n",
       "      <td>ENSMUSG00000051951.5</td>\n",
       "      <td>ENSMUST00000162897.1</td>\n",
       "      <td>Xkr4</td>\n",
       "      <td>Xkr4-203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>3206523</td>\n",
       "      <td>3215632</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000051951.5\"; transcript_id ...</td>\n",
       "      <td>ENSMUSG00000051951.5</td>\n",
       "      <td>ENSMUST00000159265.1</td>\n",
       "      <td>Xkr4</td>\n",
       "      <td>Xkr4-202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>3214482</td>\n",
       "      <td>3671498</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000051951.5\"; transcript_id ...</td>\n",
       "      <td>ENSMUSG00000051951.5</td>\n",
       "      <td>ENSMUST00000070533.4</td>\n",
       "      <td>Xkr4</td>\n",
       "      <td>Xkr4-201</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       0        1           2        3        4  5  6  7  \\\n",
       "1   chr1   HAVANA  transcript  3073253  3074322  .  +  .   \n",
       "4   chr1  ENSEMBL  transcript  3102016  3102125  .  +  .   \n",
       "7   chr1   HAVANA  transcript  3205901  3216344  .  -  .   \n",
       "10  chr1   HAVANA  transcript  3206523  3215632  .  -  .   \n",
       "13  chr1   HAVANA  transcript  3214482  3671498  .  -  .   \n",
       "\n",
       "                                                    8               gene_id  \\\n",
       "1   gene_id \"ENSMUSG00000102693.1\"; transcript_id ...  ENSMUSG00000102693.1   \n",
       "4   gene_id \"ENSMUSG00000064842.1\"; transcript_id ...  ENSMUSG00000064842.1   \n",
       "7   gene_id \"ENSMUSG00000051951.5\"; transcript_id ...  ENSMUSG00000051951.5   \n",
       "10  gene_id \"ENSMUSG00000051951.5\"; transcript_id ...  ENSMUSG00000051951.5   \n",
       "13  gene_id \"ENSMUSG00000051951.5\"; transcript_id ...  ENSMUSG00000051951.5   \n",
       "\n",
       "           transcript_id      gene_name    transcript_name  \n",
       "1   ENSMUST00000193812.1  4933401J01Rik  4933401J01Rik-201  \n",
       "4   ENSMUST00000082908.1        Gm26206        Gm26206-201  \n",
       "7   ENSMUST00000162897.1           Xkr4           Xkr4-203  \n",
       "10  ENSMUST00000159265.1           Xkr4           Xkr4-202  \n",
       "13  ENSMUST00000070533.4           Xkr4           Xkr4-201  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(133849, 8)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>6</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3073253</td>\n",
       "      <td>3074322</td>\n",
       "      <td>+</td>\n",
       "      <td>ENSMUST00000193812.1</td>\n",
       "      <td>4933401J01Rik-201</td>\n",
       "      <td>ENSMUSG00000102693.1</td>\n",
       "      <td>4933401J01Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3102016</td>\n",
       "      <td>3102125</td>\n",
       "      <td>+</td>\n",
       "      <td>ENSMUST00000082908.1</td>\n",
       "      <td>Gm26206-201</td>\n",
       "      <td>ENSMUSG00000064842.1</td>\n",
       "      <td>Gm26206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3205901</td>\n",
       "      <td>3216344</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSMUST00000162897.1</td>\n",
       "      <td>Xkr4-203</td>\n",
       "      <td>ENSMUSG00000051951.5</td>\n",
       "      <td>Xkr4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3206523</td>\n",
       "      <td>3215632</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSMUST00000159265.1</td>\n",
       "      <td>Xkr4-202</td>\n",
       "      <td>ENSMUSG00000051951.5</td>\n",
       "      <td>Xkr4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3214482</td>\n",
       "      <td>3671498</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSMUST00000070533.4</td>\n",
       "      <td>Xkr4-201</td>\n",
       "      <td>ENSMUSG00000051951.5</td>\n",
       "      <td>Xkr4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       0        3        4  6         transcript_id    transcript_name  \\\n",
       "1   chr1  3073253  3074322  +  ENSMUST00000193812.1  4933401J01Rik-201   \n",
       "4   chr1  3102016  3102125  +  ENSMUST00000082908.1        Gm26206-201   \n",
       "7   chr1  3205901  3216344  -  ENSMUST00000162897.1           Xkr4-203   \n",
       "10  chr1  3206523  3215632  -  ENSMUST00000159265.1           Xkr4-202   \n",
       "13  chr1  3214482  3671498  -  ENSMUST00000070533.4           Xkr4-201   \n",
       "\n",
       "                 gene_id      gene_name  \n",
       "1   ENSMUSG00000102693.1  4933401J01Rik  \n",
       "4   ENSMUSG00000064842.1        Gm26206  \n",
       "7   ENSMUSG00000051951.5           Xkr4  \n",
       "10  ENSMUSG00000051951.5           Xkr4  \n",
       "13  ENSMUSG00000051951.5           Xkr4  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_out = df[[0, 3, 4, 6, 'transcript_id', 'transcript_name', 'gene_id', 'gene_name']]\n",
    "print(df_out.shape)\n",
    "df_out.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "f = '/cndd2/fangming/references/gencode_vM16_transcript_annotation.bed'\n",
    "df_out.to_csv(f, sep='\\t', header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
